### ISSUE TITLE
SyntheticStore on handles via PouchDb returning empty array

### ISSUE THREAD

**Issue by sjmiles** [PolymerLabs/arcs#2440]
When I connect to this storageKey (e.g.)
```
synthetic://arc/handles/pouchdb://local/pouch/planner-launcher
```
It returns a Collection but `toList()` is empty (even though the store has values). See `planner-launcher/handles` here:

![image](https://user-images.githubusercontent.com/98045/49903323-afb13b00-fe1b-11e8-9521-8e8ed894547d.png)

Should be able to reproduce by opening 
`arcs/shells/web-shell/?log=&storage=pouchdb://local/pouch`
for any user. 

See numerous `failed to marshal arcsStore for [planner][pouchdb://local/pouch]` in console.

**Comment by mykmartin** [comment]
I think the problem might be at https://github.com/PolymerLabs/arcs/blob/master/src/runtime/storage/pouchdb/pouch-db-storage.ts#L98 - this is calling construct() from connect(), which doesn't look right. connect() should fail if the target doesn't exist, not call construct.

**Comment by mykmartin** [comment]
Also, the `$USER-launcher/handles` list isn't what synthetic storage is looking at. It connects to `$USER-launcher/arc-info` and is expecting a serialised version of the arc to exist (in an Entity-like format) under that.

**Comment by sjmiles** [comment]
> $USER-launcher/arc-info and is expecting a serialised version of the arc to exist 

Yes, I should not have jumped over that. Here is a complete dump of what's in the pouch:

![image](https://user-images.githubusercontent.com/98045/49972833-b60ce900-fee7-11e8-9363-ffe72a781f7f.png)

It looks like the serialization doesn't exist. I confirmed that `persistSerialization` is being called and there is a `set` operation for the ArcInfo, but it doesn't seem to make it into pouch.

I messed up assigning this to multiple folks. I think it's in @lindner's court right now (pouch issue?) so I'm removing Myk.
